examples/migration/migrate.py

```diff
@@ -0,0 +1,300 @@
+                    logging.debug("The task tree for {0} is:".format(full_rest_url))
+                    logging.debug("Returned JSON: {0}".format(r_treejson))
+                else:
+                    logging.debug("Exception during get call, response was not set")
+                    logging.debug("Unable to get the task tree for {0}".format(full_rest_url))
+            return(task_state, task_status)
+        except ValueError as e:
+            raise Exception('Error getting the JSON results from the task. Originating request on URL: {0}. Exception: {1}'.format(calling_url, e))
+        except Exception as e:
+            raise Exception('Error in polling for the task. Originating request on URL: {0}. Exception: {1}'.format(calling_url, e))
+
+    def get_migratable_device(self):
+        '''Method to get list of migratable device .
+
+        Return
+        ------
+        list containing server hardware that are eligible for migration:
+
+        '''
+
+        url = '/rest/migrate/migratable-server-hardware/'
+        payload = {"ipAddress": self.migrate_source_ip,
+                   "username": self.migrate_source_username, "password": self.migrate_source_password}
+
+        (migrateable_success, migrateable_resp, migrateable_json_response, _) = self.appliance_request(
+            request_type='POST', url=url, secure=True, payload=payload)
+        if migrateable_success:
+            logging.info('migrateable accepted.')
+        else:
+            raise Exception('migratable device listing failed. Status: {0}. JSON Response: {1}'
+                            .format(migrateable_resp.status_code, json.dumps(migrateable_json_response, sort_keys=True,
+                                                                             indent=4, separators=(',', ': '))))
+
+        sh_list = migrateable_json_response['serverHardware']
+
+        return sh_list
+
+    def do_migrate(self):
+        '''Method to perform the migration of selected harware .
+
+        '''
+        url = '/rest/migrate'
+        payload = {"ipAddress": self.migrate_source_ip, "username": self.migrate_source_username,
+                   "password": self.migrate_source_password, "uri": self.migrate_source_hardware_uri}
+
+        migrate_json_response = {}
+        (migrate_success, migrate_resp, migrate_json_response, _) = self.appliance_request(
+            request_type='POST', url=url, secure=True, payload=payload)
+        if migrate_success:
+            logging.info('migration successful.')
+        else:
+            raise Exception('migration failed. Status: {0}. JSON Response: {1}'.
+                            format(migrate_resp.status_code, json.dumps(migrate_json_response, sort_keys=True, indent=4, separators=(',', ': '))))
+
+
+config = {
+    "appliance_ip": "<appliance_ip>",
+    "source_appliance_ip": "<source_appliance_ip>",
```

Comment:
can we have something like
source_appliance: {
                  username: "",
                  password: ""
},
target_appliance: {
              "user_name": "",
              "password": ""
}